ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -118,49 +125,123 @@ ACP-103 introduced the following variables for calculating the gas price:
+$$
+\begin{align}
+\Delta{t} &~:= \max\left(t_e, t_b\right) - t_e \\
+t_e &~:= t_e + \Delta{t} \\
+x &~:= \max\left(x - T \cdot \Delta{t}, 0\right) \\
+\end{align}
+$$
 
-$$x := \max\left(x-\frac{s}{p}, 0\right)$$
+The block is then executed with the gas price calculated by the current value of $x$.
 
-and the gas price is updated before the next chunk.
-With the current $p = 2$, half of all gas consumed is added to excess while half of all surplus is deducted.
+After executing a block which was charged $g_C$ gas, the executor's timestamp and excess is updated:
+
+$$
+\begin{align}
+\Delta{t} &~:= \frac{g_C}{R} \\
+t_e &~:= t_e + \Delta{t} \\
+x &~:= x + \Delta{t} \cdot (R - T) \\
+\end{align}
+$$
 
 > [!NOTE]
-> The ACP-103 concept of a leaky gas bucket with capacity $C$ is an emergent phenomenon in SAE.
+> The update rule here assumes that $t_e$ is a timestamp that can track the passage of time both by gas and by wall-clock time. $\frac{g_C}{R}$ MUST NOT be simply rounded. Rather, the gas accumulation should be left as a fraction.
 
-### JSON RPC methods
+$t_e$ is now this block's execution timestamp.
 
-Although asynchronous execution decouples the transactions and receipts recorded by a specific block, APIs MUST NOT alter their behavior to mirror this.
-In particular, the API method `eth_getBlockReceipts` MUST return the receipts corresponding to the block's transactions, not the receipts settled in the block.
+### Handling gas target changes
 
-To achieve this, we first define the _executed_ and _settled_ states for blocks.
-A block is initially _accepted_ by consensus, enqueuing its tranche of transactions.
-Once the last transaction in the tranche has been executed, the block itself is considered to be executed.
-When the same transaction is settled then so too is the block considered settled.
+When a block is produced which modifies $T$, both the consensus thread and the execution thread will update to the modified $T$ after their own handling of the block.
 
-#### Named blocks
+For example, restrictions of the queue size MUST be calculated based on the parent block's $T$.
 
-The Ethereum Mainnet APIs allow for retrieving blocks by named parameters that the API server resolves based on their consensus mechanism.
-Other than the _earliest_ (genesis) named block, which MUST be interpreted in the same manner, all other named blocks are mapped to SAE in terms of the execution status of all block transactions and MUST be interpreted as follows:
+Similarly, the time spent executing a block MUST be calculated based on the parent block's $T$.
 
- * _pending_: the most recently _accepted_ block;
- * _latest_: the block that was most recently _executed_;
- * _safe_ and _finalized_: the block that was most recently _settled_.
+### Block settlement
+
+For a _proposed_ block which includes timestamp $t_b$, all ancestors whose execution timestamp $t_e$ is $t_e + \tau \leq t_b$ are considered settled.
```

Comment:
```suggestion
For a _proposed_ block that includes timestamp $t_b$, all ancestors whose execution timestamp $t_e$ is $t_e \leq t_b - \tau$ are considered settled.
Note that $t_e$ is not an integer as it tracks fractional seconds with gas consumption, which is not the case for $t_b$.
```

Reply:
In the code, I would get nervous about underflow, but for the spec it's fine 😉